get_data_meteorological_multiworker.py

```diff
@@ -0,0 +1,249 @@
+import os
+import sys
+import time
+import traceback
+import json
+import random
+import csv
+import queue
+import logging
+from multiprocessing import Pool
+import multiprocessing
+
+import requests
+
+
+TOKEN_FILE = "tokens.listing"
+cdo_tokens = multiprocessing.Queue()
+STATION_LOOKUP={}
+
+
+logger = logging.getLogger("noaa_climate_pull")
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+handlerF = logging.FileHandler("noaa_climate_pull.log")
+handlerF.setFormatter(formatter)
+handlerF.setLevel(logging.DEBUG)
+
+handlerC = logging.StreamHandler()
+handlerC.setFormatter(formatter)
+handlerC.setLevel(logging.INFO)
+
+# Add the handlers to the logger
+logger.addHandler(handlerF)
+logger.addHandler(handlerC)
+
+
+def get_timeseries_batched(entity_desc, datasetid, stn_id, year_start, year_end, request_uri, out_file_json, out_file_csv, field_list, extra_params=None):
+
+    result_count = 1000000 # does not matter atm, offset of 0 will always dominate
+    current_offset = 0
+    result_limit = 1000
+    no_results = False
+
+    list_obtained = []
+
+    license_key = cdo_tokens.get(block=True)
+    logger.debug(f"Using License key {license_key}")
+
+    stn_name = STATION_LOOKUP[stn_id]["stn_name"]
+    logger.info(f"Pulling for station {stn_id}, readings from {year_start}-{year_end}, location: {stn_name}")
+
+    try:
+
+        for year_current in range(year_start, year_end+1):
+            no_results = False
+            current_offset = 0
+
+            head = {'token': license_key}
+
+            while not no_results and (current_offset < result_count):
+                time.sleep(1)
+                request_uri_parameterized = f"{request_uri}?limit={result_limit}&offset={current_offset}&startdate={year_current}-01-01&enddate={year_current}-12-31&datasetid={datasetid}&stn_id={stn_id}"
```

Comment:
Rather than having a string formatted, we can use a dict and pass it in using the `param` argument/

For instance
```
params = {
    'limit': 100,
    'offset': current_offsett,
   'startdate': year_current
    ....etc
}

response = response.get(request_uri, params=params)
```

For more information, you can see it [here](https://requests.kennethreitz.org/en/master/user/quickstart/#passing-parameters-in-urls)

This will make it easier to read and maintain.